src/widget/wfindonwebmenu.h

```diff
@@ -1,10 +1,8 @@
 #pragma once
 
-#include <QAction>
 #include <QMenu>
-#include <QUrlQuery>
 
-class Track;
+#include "track/track.h"
```

Comment:
```suggestion
```
this is quite a heavy include and I see no reason to include it here.

Reply:
I have included
> #include "track/track.h"
into all of the service menus. I think including them in every service classes is heavy too. I have deleted them, there are no problems after deleting them.

But I couldn't delete the line 5 on wfindonwebmenu.h, because this time, service classes are giving me an error. Is there any other way to solve this issue?

Reply:
Headers are only included once in each translation unit (each cpp file being compiled). Since you included them in `wfindonwebmenu.h`, which you then included in each service-menu header. They all automatically included `track.h` too. This is called a transitive dependency. Since every header gets only `#include`d once (or rather there is a thing called "include guards" (which we have in the form of `#pragma once`) for headers which takes care of that), when you included them in service menu, they actually didnt get included because they already were from the `#include wfindonwebmenu.h`. So the solution here is to eliminate the transitive dependency, `#include track/track.h` in the service menus and delete the include in `wfindonwebmenu.h`.

Reply:
Thanks for this detailed explanation. I am just asking for to be make sure before my next commit. The solution might be something that is so simple, I'm sorry to ask something like that.

I have added `#include track/track.h` in service menus. After I have tried to delete the `#include track/track.h` in `wfindonwebmenu.h`. I am having an error related to Track. 

Which is `'Track' does not name a type`. I have looked at the code base, I have seen the `track_decl.h` and I tried it, still an error `invalid use of incomplete type` occurs. I have used forward declaration still the same.

I can try to move this function out of the class, maybe in WTrackMenu, But when I first started to work on it, we made decision about not bothering the WTrackMenu. 

I'm a bit confused. What should I do?

Reply:
I can't really help you with the error message unless you include the files and linenumbers they occur on. In general, use `track_decl.h` or forward declaration in the headers, use the full `track.h` in the implementation (`.cpp`) files.

Reply:
Ohh, okay. So instead of including `"track/track.h"` in `wfindonwebmenu.h` i can include `"track/track.h"` in `wfindonwebmenu.cpp` right?

Reply:
Exactly (if I'm not mistaken :sweat_smile:), Though since `Track` actually does not occur in `wfindonwebmenu.h` you don't even need to forward declare it in the header.

Reply:
Thanks a lot for the help :octopus:

I tried to delete the forward declare, it gives error. So It's better to keep it there :sweat_smile: . 

In a bit I will commit the header includes.

Reply:
Ah yes, I didn't see there was `static bool hasEntriesForTrack(const Track& track)` which obviously needs that forward-declaration.

Reply:
I have pushed the include changes with a little commit. I hope that little commit is matched with the requested changes.